### ISSUE TITLE
"disabled" property of constraints is ignored

### ISSUE THREAD

**Issue by mephinet** [perl5-dbi/DBD-Oracle#23]
Oracle allows you to disable constraints, cf: https://docs.oracle.com/cd/B19306_01/server.102/b14200/clauses002.htm#i1002349 . DBD::Oracle's `primary_key_info` and `foreign_key_info`, however, ignore this flag when dumping the `ALL_CONSTRAINTS` table and thereby return disabled primary keys.

**Comment by mjegh** [comment]
You could try the following:

```sql
diff --git a/lib/DBD/Oracle.pm b/lib/DBD/Oracle.pm
index 09a0022..78ceac4 100644
--- a/lib/DBD/Oracle.pm
+++ b/lib/DBD/Oracle.pm
@@ -535,6 +535,7 @@ SELECT *
      AND p.TABLE_NAME      = c.TABLE_NAME
      AND p.CONSTRAINT_NAME = c.CONSTRAINT_NAME
      AND p.CONSTRAINT_TYPE = 'P'
+     AND p.STATUS = 'ENABLED'
 )
  WHERE TABLE_SCHEM = ?
    AND TABLE_NAME  = ?
@@ -586,6 +587,7 @@ SELECT *
      AND uk.CONSTRAINT_NAME  = fk.R_CONSTRAINT_NAME
      AND uk.OWNER            = fk.R_OWNER
      AND uc.POSITION         = fc.POSITION
+     AND fk.STATUS           = 'ENABLED'
 )
  WHERE 1              = 1
 SQL
```

**Comment by mjegh** [comment]
I sent a pull request for this.

**Comment by mephinet** [comment]
Great, thanks, mjegh!

**Comment by djzort** [comment]
@mjegh this is yours :)

**Comment by mjegh** [comment]
@djzort I know. Not sure I got any feedback as to whether this actually worked for @mephinet, just a "Great". If @mephinet says it works, fine, we'll merge but it is potentially a change in behaviour so at least will need to be noted carefully.

**Comment by djzort** [comment]
(probably needs a test too)

**Comment by mephinet** [comment]
I'll see if I'm able to test this change - gimme one week...

**Comment by mephinet** [comment]
@mjegh  sorry, took some time to get a working Perl + Oracle environment again - but I can confirm that https://github.com/perl5-dbi/DBD-Oracle/pull/25 works correctly. As it was already published as part of 1.75, I'm closing the issue.

**Comment by mjegh** [comment]
No problems @mephinet . We got there in the end.
